### PR TITLE
Enum tests

### PR DESCRIPTION
**Pull request by justkawal**
## Description
* Enums testing:
    - `enum: ['Value1', 'Value2', 'Value3']`
    - `enum: { 'complex': {'index': 'u8' } }`

## Why?
Resolves issues: #47 

## Testing Plan 
- `dart test`

### CODE REVIEW

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    // fetching the parsed types from `Json` to `Type`
+    final types = registry.getTypes();
+
+    // Initializing Scale-Codec object
+    final codec = Codec(types);
+
+    group('Encode ComplexEnum:', () {
+      test(
+          'When {\'Plain\': \'scale-codec\'} is encoded then it produces \'0x002c7363616c652d636f646563\'.',
```
Comment (gabrielokura) [inline packages/polkadart_scale_codec/test/test_types/enum_test.dart:111]: I know 'scale-codec' is not a number, but the idea here is basically the same of using 'magical numbers', right? @pastre

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart [hunk 2]
```diff
@@ -0,0 +1,178 @@
+import 'package:polkadart_scale_codec/polkadart_scale_codec.dart';
+import 'package:test/test.dart';
+
+void main() {
+  {
+    // Creates the registry for parsing the types
+    final registry = TypeRegistry(
+      types: {
+        'FavouriteColorEnum': {
+          '_enum': ['Red', 'Orange']
```
Comment (gabrielokura) [inline packages/polkadart_scale_codec/test/test_types/enum_test.dart:10]: Should we test simple enums with integers and others type? @leonardocustodio

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+        expect(() => codec.encode(typeIndex, value),
+            throwsA(isA<UnknownVariantException>()));
+      });
+    });
+  }
+
+  {
+    // Creates the registry for parsing the types
+    final registry = TypeRegistry(
```
Comment (gabrielokura) [inline packages/polkadart_scale_codec/test/test_types/enum_test.dart:83]: Couldn't we declare registry and codec once in the file? You could use the same instance to encode and decode values

**Review by leonardocustodio — APPROVED** [review]

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart (hunk 3 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart]: Done ✅

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart (hunk 2 above)
Comment (leonardocustodio) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart]: Yes

Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart]: Done added

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart (hunk 1 above)
Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart]: changed description.

**Comment by justkawal** [comment]
@gabrielokura Can you please review ?

File: packages/polkadart_scale_codec/test/test_types/enum_test.dart [hunk 4]
```diff
@@ -0,0 +1,216 @@
+import 'package:polkadart_scale_codec/polkadart_scale_codec.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // Creates the registry for parsing the types
+  final registry = TypeRegistry(
+    types: {
+      'FavouriteColorEnum': {
+        '_enum': ['Red', 'Orange']
```
Comment (gabrielokura) [inline packages/polkadart_scale_codec/test/test_types/enum_test.dart:9]: i mean, `_enum` can have values of any type? Like: 
```
'_enum': [1000, true, ['Red', 'Orange']]
```

Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart:9]: No, It can't

Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart:9]: Like:
```
// Simple type enum
'FavouriteColorEnum': { '_enum': ['Red', 'Orange'] }

// Complex type enum with mixture of values.
'OrderEnum': {
   '_enum': {
        'Plain': {
            'index': 'u8',
        },
        'Custom': {
            'index': 'u8',
            'booleanValue': 'bool',
        },
   },
},
```

In 'OrderEnum', you can either initiate 'Plain' or 'Custom' as it is under '_enum'.

When initiating both under the HashMap then the very first key of hashmap is considered and the rest is ignored as a nature of enum.

Comment (justkawal) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart:9]: > https://github.com/paritytech/parity-scale-codec-ts
> 
> Parity supports booleans and integers too

Under `[]` are named enums.

Under `{}` are parameterised enums.

Comment (gabrielokura) [reply packages/polkadart_scale_codec/test/test_types/enum_test.dart:9]: Yes, i'm talking about named enums with integers, not only strings

**Review by gabrielokura — APPROVED** [review]

**Pull request merged**